lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -296,6 +300,19 @@ defmodule Mix.Tasks.Deps.Compile do
     true
   end
 
+  defp do_gleam(%Mix.Dep{opts: opts} = dep, config) do
+    lib = Path.join(Mix.Project.build_path(), "lib")
+    out = opts[:build]
+    package = opts[:dest]
+
+    command =
+      "gleam compile-package --target erlang " <>
+        "--package  \"#{package}\" --out \"#{out}\" --lib \"#{lib}\""
```

Comment:
I think the command can be a tuple like this:

    {"gleam", ["compile-package", "--target", "erlang", ...]}

Which would have the benefit of not needing escape.